mm2src/coins/tendermint/tendermint_wasm_rpc.rs

```diff
@@ -0,0 +1,98 @@
+use cosmrs::tendermint::abci::Transaction;
+use derive_more::Display;
+use http::uri::InvalidUri;
+use http::{StatusCode, Uri};
+use mm2_net::transport::SlurpError;
+use mm2_net::wasm_http::FetchRequest;
+use std::str::FromStr;
+pub use tendermint_rpc::endpoint::abci_query::Request as AbciRequest;
+use tendermint_rpc::endpoint::{abci_info, broadcast};
+use tendermint_rpc::error::Error as TendermintRpcError;
+use tendermint_rpc::request::SimpleRequest;
+use tendermint_rpc::Response;
+
+#[derive(Debug)]
+pub(super) struct HttpClient {
+    uri: String,
+}
+
+#[derive(Debug, Display)]
+pub(super) enum HttpClientInitError {
+    InvalidUri(InvalidUri),
+}
+
+impl From<InvalidUri> for HttpClientInitError {
+    fn from(err: InvalidUri) -> Self { HttpClientInitError::InvalidUri(err) }
+}
+
+#[derive(Debug, Display)]
+pub(super) enum PerformError {
+    TendermintRpc(TendermintRpcError),
+    Slurp(SlurpError),
+    #[display(fmt = "Request failed with status code {}, response {}", status_code, response)]
+    StatusCode {
+        status_code: StatusCode,
+        response: String,
+    },
+}
+
+impl From<SlurpError> for PerformError {
+    fn from(err: SlurpError) -> Self { PerformError::Slurp(err) }
+}
+
+impl From<TendermintRpcError> for PerformError {
+    fn from(err: TendermintRpcError) -> Self { PerformError::TendermintRpc(err) }
+}
+
+impl HttpClient {
+    pub(super) fn new(url: &str) -> Result<Self, HttpClientInitError> {
+        Uri::from_str(url)?;
+        Ok(HttpClient { uri: url.to_owned() })
+    }
+
+    pub(super) async fn perform<R>(&self, request: R) -> Result<R::Response, PerformError>
+    where
+        R: SimpleRequest,
+    {
+        let request_str = request.into_json();
+        let (status_code, response_str) = FetchRequest::post(&self.uri)
+            .cors()
+            .body_utf8(request_str)
+            .header("Accept", "application/json")
+            .header("Content-Type", "application/json")
```

Comment:
Can we create these values as constant in `wasm_http` module ? Since we already have them from hyper for native http mode, it would be nice to have the same aspect in wasm http mode.

Reply:
Sure, will do 🙂